x-pack/legacy/plugins/siem/public/components/ml_popover/api.tsx

```diff
@@ -0,0 +1,142 @@
+ * @param configTemplate - name of configTemplate to setup
+ * @param indexPatternName - default index pattern configTemplate should be installed with
+ * @param groups - list of groups to add to jobs being installed
+ * @param prefix - prefix to be added to job name
+ */
+export const setupMlJob = async ({
+  configTemplate,
+  indexPatternName = 'auditbeat-*',
+  groups = ['siem'],
+  prefix = '',
+}: MlSetupArgs) => {
+  const response = await fetch(`/api/ml/modules/setup/${configTemplate}`, {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      prefix,
+      groups,
+      indexPatternName,
+      startDatafeed: false,
+      useDedicatedIndex: false,
+    }),
+    headers: {
+      'kbn-system-api': 'true',
+      'Content-Type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+    },
+  });
+  return await response.json();
+};
+
+/**
+ * Starts the given dataFeedIds
+ *
+ * @param datafeedIds
+ */
+export const startDatafeeds = async (datafeedIds: string[]) => {
+  const response = await fetch('/api/ml/jobs/force_start_datafeeds', {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      datafeedIds,
+    }),
+    headers: {
+      'kbn-system-api': 'true',
+      'Content-Type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+    },
+  });
+  return await response.json();
+};
+
+/**
+ * Stops the given dataFeedIds and sets the corresponding Job's jobState to closed
+ *
+ * TODO: Error Handling
+ *
+ * @param datafeedIds
+ */
+export const stopDatafeeds = async (datafeedIds: string[]) => {
+  const stopDatafeedsResponse = await fetch('/api/ml/jobs/stop_datafeeds', {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      datafeedIds,
+    }),
+    headers: {
+      'kbn-system-api': 'true',
+      'Content-Type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+    },
+  });
+
+  const stopDatafeedsResponseJson = await stopDatafeedsResponse.json();
+
+  const datafeedPrefix = 'datafeed-';
+  const closeJobsResponse = await fetch('/api/ml/jobs/close_jobs', {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      jobIds: datafeedIds.map(dataFeedId =>
+        dataFeedId.startsWith(datafeedPrefix)
+          ? dataFeedId.substring(datafeedPrefix.length)
+          : dataFeedId
+      ),
+    }),
+    headers: {
+      'kbn-system-api': 'true',
+      'Content-Type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+    },
+  });
+
+  return [stopDatafeedsResponseJson, await closeJobsResponse.json()];
+};
+
+/**
+ * Fetches Job Details for given jobIds
+ *
+ * @param jobIds
+ */
+export const jobsSummary = async (jobIds: string[]) => {
+  const response = await fetch('/api/ml/jobs/jobs_summary', {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({ jobIds }),
+    headers: {
+      'kbn-xsrf': chrome.getXsrfToken(),
+    },
```

Comment:
For headers the two things we have to ensure are included looks to be:

* kbn-system-api: 'true',
* kbn-version: config.kbnVersion

And that will be consistent with their API calls.

Reply:
Thanks! I've updated all API calls to use these headers:

``` ts
headers: {
  'content-type': 'application/json',
    'kbn-system-api': 'true',
    'kbn-xsrf': chrome.getXsrfToken(),
    'kbn-version': config.kbnVersion,
},
```